app/models/user.rb

```diff
@@ -85,17 +72,42 @@ def can_access_evss?
     edipi.present? && ssn.present? && participant_id.present?
   end
 
+  def edipi
+    select_source_id(:edipi)
+  end
+
+  def icn
+    select_source_id(:icn)
+  end
+
+  def participant_id
+    select_source_id(:vba_corp_id)
+  end
+
+  def mhv_correlation_id
+    mhv_correlation_ids.first
+  end
+
+  def mvi
+    return nil unless loa3?
+    @mvi ||= Mvi.from_user(self).fetch
+  end
+
   private
 
   def mhv_correlation_ids
     return @mhv_correlation_ids if @mhv_correlation_ids
     ids = mvi&.dig(:mhv_ids)
     ids = [] unless ids
-    @mhv_correlation_ids = ids.map { |mhv_id| mhv_id.split('^')&.first }.compact
-    @mhv_correlation_ids
+    ids.map { |mhv_id| mhv_id.split('^')&.first }.compact
```

Comment:
Looks like you've dropped the memoization here. If you don't want it to memoize, you could remove the first line of the method.